lib/rubocop/cop/rspec/described_class.rb

```diff
@@ -38,27 +52,42 @@ def on_block(node)
           describe, described_class, body = described_constant(node)
           return unless top_level_describe?(describe)
 
-          find_constant_usage(body, described_class) do |match|
-            add_offense(match, :expression, format(MSG, match.const_name))
+          @described_class = described_class
```

Comment:
@backus I completely agree, but the alternative is in the autocorrect method to travel the nodes up to the top-level describe, in order to find what should be used. I can attempt such implementation of course, but if there are any other alternatives, I would like to explore them. Any ideas?